Build wheels for Python 3.14

I have followed the Manim set up steps in the documentation, however I run into troubles. I think I have manage to get around some problems however I'm now stuck. I get the error saying: 

```
`Resolved 39 packages in 1.03s
      Built moderngl==5.12.0
      Built glcontext==3.0.0                                                                                              
  × Failed to build `manimpango==0.6.0`                                                                                   
  ├─▶ The build backend returned an error                                                                                 
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit code: 1)                                                                                                                                                                                   [stdout]                                                                                                            
      running bdist_wheel                                                                                                 
      running build                                                                                                       
      running build_py
      creating build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\register_font.py -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\utils.py -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\_version.py -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\__init__.py -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\cairo.pxd -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\cmanimpango.pxd -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\glib.pxd -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\pango.pxd -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\_register_font.pxd -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\cmanimpango.pyx -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\enums.pyx -> build\lib.win-amd64-cpython-314\manimpango
      copying manimpango\_register_font.pyx -> build\lib.win-amd64-cpython-314\manimpango
      running build_ext
      building 'manimpango.cmanimpango' extension
      creating
      build\temp.win-amd64-cpython-314\Release\Users\Hannes\AppData\Local\uv\cache\sdists-v9\pypi\manimpango\0.6.0\xYIFcPm1XRSuCl89dpvTq\src\manimpango
      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.44.35207\bin\HostX86\x64\cl.exe" /c
      /nologo /O2 /W3 /GL /DNDEBUG /MD -DUNICODE=1 -IC:\Users\Hannes\AppData\Local\uv\cache\builds-v0\.tmp3HlNjr\include  
      -IC:\Users\Hannes\AppData\Roaming\uv\python\cpython-3.14.0-windows-x86_64-none\include
      -IC:\Users\Hannes\AppData\Roaming\uv\python\cpython-3.14.0-windows-x86_64-none\Include "-IC:\Program
      Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.44.35207\include" "-IC:\Program Files
      (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows
      Kits\10\include\10.0.26100.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\um"
      "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\shared" "-IC:\Program Files (x86)\Windows
      Kits\10\\include\10.0.26100.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\cppwinrt"    
      /TcC:\Users\Hannes\AppData\Local\uv\cache\sdists-v9\pypi\manimpango\0.6.0\xYIFcPm1XRSuCl89dpvTq\src\manimpango\cmanimpango.c
      /Fobuild\temp.win-amd64-cpython-314\Release\Users\Hannes\AppData\Local\uv\cache\sdists-v9\pypi\manimpango\0.6.0\xYIFcPm1XRSuCl89dpvTq\src\manimpango\cmanimpango.obj
      -DCAIRO_WIN32_STATIC_BUILD
      cmanimpango.c
      C:\Users\Hannes\AppData\Local\uv\cache\sdists-v9\pypi\manimpango\0.6.0\xYIFcPm1XRSuCl89dpvTq\src\manimpango\cmanimpango.c(1259):
      fatal error C1083: Cannot open include file: 'cairo.h': No such file or directory

      [stderr]
      <string>:99: RuntimeWarning: pkg-config doesn't exists or doesn't seem to work
      We assume that you give the compiler flags using
      corresponding environment variables.
      C:\Users\Hannes\AppData\Local\uv\cache\builds-v0\.tmp3HlNjr\Lib\site-packages\setuptools\dist.py:759:
      SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: MIT License

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      C:\Users\Hannes\AppData\Local\uv\cache\builds-v0\.tmp3HlNjr\Lib\site-packages\setuptools\dist.py:483:
      SetuptoolsDeprecationWarning: Cannot find any files for the given pattern.
      !!

              ********************************************************************************
              Pattern 'LICENSE.*' did not match any files.

              By 2026-Mar-20, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
              ********************************************************************************

      !!
        for path in sorted(cls._find_pattern(pattern, enforce_match))
      error: command 'C:\\Program Files (x86)\\Microsoft Visual
      Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.44.35207\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      <string>:51: RuntimeWarning: ManimPango 0.6.0 does not support Python 3.14 and does not provide prebuilt Windows    
      binaries. We do not recommend building from source on Windows.

      hint: This usually indicates a problem with the package or the build environment.
  help: If you want to add the package regardless of the failed resolution, provide the `--frozen` flag to skip locking   
        and syncing.`
```

The way I interpret the error is that I have to use another Python version so that it can run with ManimPango. Do I really have to use another Python version? If I can use Python 3.14, how would I go about it?

Best
Hannes

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -31,7 +31,7 @@ jobs:
             platform_id: macosx_arm64
             architecture: arm64
     env:
-      CIBW_BUILD: cp39-${{ matrix.platform_id }} cp310-${{ matrix.platform_id }} cp311-${{ matrix.platform_id }} cp312-${{ matrix.platform_id }} cp313-${{ matrix.platform_id }}
+      CIBW_BUILD: cp39-${{ matrix.platform_id }} cp310-${{ matrix.platform_id }} cp311-${{ matrix.platform_id }} cp312-${{ matrix.platform_id }} cp313-${{ matrix.platform_id }} cp314-${{ matrix.platform_id }}
       CIBW_SKIP: pp* cp36* cp37* cp38*
       CIBW_BEFORE_BUILD_MACOS: "python packing/download_pango_macos.py ${{ matrix.architecture }}"
       CIBW_BEFORE_BUILD_WINDOWS: "python packing/download_dlls.py"
@@ -77,7 +77,7 @@ jobs:
       fail-fast: false
       matrix:
         architecture: [x64, x86]
-        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13', '3.14']
         include: [
           {platform_id: win_amd64, architecture: x64},
           {platform_id: win32, architecture: x86},
@@ -118,7 +118,7 @@ jobs:
       fail-fast: false
       matrix:
         architecture: [x64, arm64]
-        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13', '3.14']
         include: [
           {platform_id: macosx_x86_64, architecture: x64, os: macos-13},
           {platform_id: macosx_arm64, architecture: arm64, os: macos-14},

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -36,7 +36,7 @@ jobs:
       CIBW_BEFORE_BUILD_MACOS: "python packing/download_pango_macos.py ${{ matrix.architecture }}"
       CIBW_BEFORE_BUILD_WINDOWS: "python packing/download_dlls.py"
       CIBW_ENVIRONMENT_WINDOWS: "PKG_CONFIG_PATH='C:\\cibw\\vendor\\lib\\pkgconfig'"
-      CIBW_ENVIRONMENT_MACOS: "PKG_CONFIG_PATH='/Users/runner/pangobuild/lib/pkgconfig'"
+      CIBW_ENVIRONMENT_MACOS: "PKG_CONFIG_PATH='/Users/runner/pangobuild/lib/pkgconfig' MACOSX_DEPLOYMENT_TARGET='10.13'"
       CIBW_TEST_REQUIRES: pytest pytest-cov
       CIBW_TEST_COMMAND: "bash {project}/packing/test_wheels.sh {project}"
     steps:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -64,7 +64,7 @@ jobs:
         env:
           PKG_CONFIG_PATH: "C:\\cibw\\vendor\\lib\\pkgconfig"
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
           path: ./wheelhouse/*.whl
           name: wheels-${{ runner.os }}-${{ matrix.platform_id }}
@@ -92,7 +92,7 @@ jobs:
           architecture: ${{ matrix.architecture }}
           allow-prereleases: true
 
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
           name: wheels-${{ runner.os }}-${{ matrix.platform_id }}
           path: ~/wheelhouse
@@ -133,7 +133,7 @@ jobs:
           architecture: ${{ matrix.architecture == 'arm64' && null || matrix.architecture }}
           allow-prereleases: true
 
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
           name: wheels-${{ runner.os }}-${{ matrix.platform_id }}
           path: ~/wheelhouse
@@ -177,7 +177,7 @@ jobs:
           python -m pip install dist/*.tar.gz
 
       - name: Store artifacts
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           path: dist/*.tar.gz
           name: manimpango-src
@@ -194,7 +194,7 @@ jobs:
       contents: write
     if: github.event_name== 'release'
     steps:
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
           path: downloads/
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -32,7 +32,7 @@ jobs:
             architecture: arm64
     env:
       CIBW_BUILD: cp39-${{ matrix.platform_id }} cp310-${{ matrix.platform_id }} cp311-${{ matrix.platform_id }} cp312-${{ matrix.platform_id }} cp313-${{ matrix.platform_id }} cp314-${{ matrix.platform_id }}
-      CIBW_SKIP: pp* cp36* cp37* cp38*
+      CIBW_SKIP: pp* cp38*
       CIBW_BEFORE_BUILD_MACOS: "python packing/download_pango_macos.py ${{ matrix.architecture }}"
       CIBW_BEFORE_BUILD_WINDOWS: "python packing/download_dlls.py"
       CIBW_ENVIRONMENT_WINDOWS: "PKG_CONFIG_PATH='C:\\cibw\\vendor\\lib\\pkgconfig'"
@@ -60,7 +60,7 @@ jobs:
           cp packing/LICENSE.bin .
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@v2.20.0
+        uses: pypa/cibuildwheel@v3.2.1
         env:
           PKG_CONFIG_PATH: "C:\\cibw\\vendor\\lib\\pkgconfig"
 

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -18,7 +18,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14"]
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}
@@ -70,7 +70,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-2022]
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14"]
         architecture: ["x64", "x86"]
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -58,7 +58,7 @@ jobs:
       - uses: codecov/codecov-action@v3
         with:
           file: ./.coverage/coverage.xml
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
           name: test-artifacts-${{matrix.os}}-${{matrix.python-version}}
           path: .pytest_temp/
@@ -99,7 +99,7 @@ jobs:
       - name: Run tests
         run: |
           pytest -s
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
           name: test-artifacts-${{matrix.os}}-${{matrix.python-version}} ${{matrix.architecture}}
           path: .pytest_temp/

--- a/.readthedocs.yaml
+++ b/.readthedocs.yaml
@@ -14,4 +14,7 @@ python:
       - method: pip
         path: .
 
+sphinx:
+  configuration: docs/conf.py
+
 formats: all

--- a/packing/build_pango_tests.sh
+++ b/packing/build_pango_tests.sh
@@ -2,7 +2,7 @@
 # build and install pango
 set -e
 
-PANGO_VERSION=1.54.0
+PANGO_VERSION=1.56.4
 
 FILE_PATH=$PWD
 PREFIX="$HOME/pangoprefix"

--- a/packing/download_dlls.py
+++ b/packing/download_dlls.py
@@ -10,7 +10,7 @@ import zipfile
 from pathlib import Path
 from urllib.request import urlretrieve as download
 
-PANGO_VERSION = "1.56.4-v1"
+PANGO_VERSION = "1.56.4-v2"
 
 
 def get_platform():

--- a/packing/download_dlls.py
+++ b/packing/download_dlls.py
@@ -10,7 +10,7 @@ import zipfile
 from pathlib import Path
 from urllib.request import urlretrieve as download
 
-PANGO_VERSION = "1.56.4"
+PANGO_VERSION = "1.56.4-v1"
 
 
 def get_platform():

--- a/packing/download_dlls.py
+++ b/packing/download_dlls.py
@@ -10,7 +10,7 @@ import zipfile
 from pathlib import Path
 from urllib.request import urlretrieve as download
 
-PANGO_VERSION = "1.54.0-v3"
+PANGO_VERSION = "1.56.4"
 
 
 def get_platform():

--- a/packing/download_pango_macos.py
+++ b/packing/download_pango_macos.py
@@ -10,7 +10,7 @@ import zipfile
 from pathlib import Path
 from urllib.request import urlretrieve as download
 
-PANGO_VERSION = "1.54.0-v3"
+PANGO_VERSION = "1.56.4"
 
 logging.basicConfig(format="%(levelname)s - %(message)s", level=logging.DEBUG)
 

--- a/packing/download_pango_macos.py
+++ b/packing/download_pango_macos.py
@@ -10,7 +10,7 @@ import zipfile
 from pathlib import Path
 from urllib.request import urlretrieve as download
 
-PANGO_VERSION = "1.56.4"
+PANGO_VERSION = "1.56.4-v1"
 
 logging.basicConfig(format="%(levelname)s - %(message)s", level=logging.DEBUG)
 

--- a/packing/download_pango_macos.py
+++ b/packing/download_pango_macos.py
@@ -10,7 +10,7 @@ import zipfile
 from pathlib import Path
 from urllib.request import urlretrieve as download
 
-PANGO_VERSION = "1.56.4-v1"
+PANGO_VERSION = "1.56.4-v2"
 
 logging.basicConfig(format="%(levelname)s - %(message)s", level=logging.DEBUG)
 

--- a/setup.py
+++ b/setup.py
@@ -280,7 +280,6 @@ setup(
         "Programming Language :: Python :: 3.13",
         "Development Status :: 4 - Beta",
         "Programming Language :: Python :: 3 :: Only",
-        "License :: OSI Approved :: MIT License",
         "Programming Language :: Cython",
     ],
     project_urls={

